Libraries/StyleSheet/precomputeStyle.js

```diff
@@ -126,6 +132,71 @@ function _multiplyTransform(
 }
 
 /**
+ * Takes a transform matrix and an origin, and applies a transformation to
+ * the matrix to change the transform origin. Mutates the passed in matrix.
+ */
+function _applyTransformOrigin(
+  matrix: Array<number>,
+  origin: Object
+): void {
+  origin = _normalizeOrigin(origin);
+  var translate = MatrixMath.createIdentityMatrix();
+  var untranslate = MatrixMath.createIdentityMatrix();
+  MatrixMath.reuseTranslate3dCommand(
+    translate,
+    origin.x,
+    origin.y,
+    origin.z
+  );
+  MatrixMath.reuseTranslate3dCommand(
+    untranslate,
+    -origin.x,
+    -origin.y,
+    -origin.z
+  );
+  MatrixMath.multiplyInto(matrix, translate, matrix);
+  MatrixMath.multiplyInto(matrix, matrix, untranslate);
+}
+
+/**
+ * Accepts the user passed in "transformOrigin" property, which
+ * in the future could be a something other than an object, and
+ * normalizes it to an {x, y, z} point.
+ * 
+ * Right now this function is little more than an optimized
+ * Object.assign, but in the future it could be more complicated.
+ */
+function _normalizeOrigin(input: Object): Object {
+  var output = { x: 0, y: 0, z: 0 };
+  invariant(
+      false,
```

Comment:
`typeof input === 'object'` ?
Linter says that this code is not reachable.
@lelandrichardson could you fix? That is the only noticeable thing that stops this PR from being accepted.